src/middlewares/balance.ts

```diff
@@ -236,4 +211,48 @@ export default class BalanceMiddleware implements Middleware {
     })
     return { accounts }
   }
+
+  modifyBalance (accountId: string, _amountDiff: BigNumber.Value): BigNumber {
```

Comment:
I think it would be good to prominently log something in here (at warn level, probably), so that if you look through the logs you can see whether the balance is being unexpectedly changed. Otherwise if an attacker got access to your admin API they could modify the balance without leaving any trail